lib/migration.js

```diff
@@ -166,6 +175,18 @@ module.exports = function(IBMDB) {
     // ------------------------------------------------------------------------
     sql = sql.concat(self.buildIndexes(model));
 
+    // add back the primary key index
+    // this snippet is required in here rather than the `buildIndexes` function
+    // because DB2 automatically creates an index with a primary key when a
+    // table is created.
+    // It is during the alter table process when we need to add back the primary
+    // index that was dropped.
+    if (idName) {
```

Comment:
I don't think this is quite going to build the primary key in all cases as you would expect.  See ibmdb.js lines 810 through 818 where the primary key could consist of many columns depending on the model definition.